### PR TITLE
feat: add support for Parquet options

### LINKED ISSUE #661: add support for ParquetOptions for load/external tables
This FR is to add support for some advanced Parquet options (list inference and enums) when loading data or defining an external table.

Additional information has been shared out of band with library maintainers.

### PR DESCRIPTION
**Pull request by plamut**
Closes #661.

For load jobs and external tables config.

**PR checklist:**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: google/cloud/bigquery/format_options.py
Comment (tseaver) [inline google/cloud/bigquery/format_options.py:87]: Seems like this would be simpler as:
```suggestion
        """
        return copy.deepcopy(self._properties)
```

Comment (plamut) [reply google/cloud/bigquery/format_options.py:87]: Indeed, thanks for the suggestion.

**Comment by plamut** [comment]
We still need to expose `ParquetOptions` in the `google.cloud.bigquery` namespace to be consistent with `enums`, etc.

**Comment by tseaver** [comment]
@googlebot I fixed it.

**Comment by plamut** [comment]
Exposed the config object in `google.cloud.bigquery` namespace and also made sure that `parquet_options` works well with the `options` attribute.

**Comment by simonvanderveldt** [comment]
@plamut We are currently a bit blocked because this functionality is missing from the bigquery Python library. Is there anything we can do to help this get merged and released? Maybe some testing?

File: google/cloud/bigquery/format_options.py
Comment (simonvanderveldt) [inline google/cloud/bigquery/format_options.py:25]: Any reason the properties cannot be passed to the constructor directly? That would simplify usage a bit because currently one has to instantiate the class first and then set the desired properties.

**Comment by plamut** [comment]
@simonvanderveldt Thanks for bumping this, it appears that it's incorrectly blocked by the CLA bot (and still needs one approval).

@shollyman Can you please convince the bot to change its mind? :) And add a review, if possible?

> Is there anything we can do to help this get merged and released? Maybe some testing?

If you want, you can test the BigQuery client with this PR branch, but primarily someone from Google with sufficient permissions needs to unblock the PR first.

**Comment by simonvanderveldt** [comment]
> If you want, you can test the BigQuery client with this PR branch, but primarily someone from Google with sufficient permissions needs to unblock the PR first.

Just tested it, working as expected :) We only need the list inference and using the library from this PR gives the same results as when using `bq load --parquet_enable_list_inference`.

I posted one review comment on the code btw.

File: google/cloud/bigquery/format_options.py
Comment (simonvanderveldt) [reply google/cloud/bigquery/format_options.py:25]: OK, I understand. It's not a big problem, just felt a bit clunky :)

**Review by stephaniewang526 — APPROVED** [review]

**Comment by stephaniewang526** [comment]
Looks like we're missing CLA signing from `tseaver@palladion.com`

**Comment by plamut** [comment]
But Tres is a well-known contributor? Or is it because a different email address was used?

**Comment by busunkim96** [comment]
@plamut I looks like there is no CLA for `tseaver@palladion.com`. @tseaver's CLA lists `tres.seaver@gmail.com`.

I think if you edit the `Co-authored-by: Tres Seaver <tseaver@palladion.com>` in the commit body the CLA bot will be appeased. Please holler if that doesn't work.

**Comment by plamut** [comment]
That trick with editing the commit message worked, great! Since we also have an approval now, I'll merge and release this today.

**Comment by plamut** [comment]
@busunkim96  As for the CLA bot, isn't this a bit fragile? Theoretically one could take another person's work posted as a suggestion, edit out the line in the commit message linking to that person, and then merge a non-CLA code into the project?

We can discuss this offline.

**Comment by plamut** [comment]
@simonvanderveldt The changes have just been released, happy coding!

**Comment by simonvanderveldt** [comment]
@plamut Awesome! Thanks a lot for the quick merge and release!

**Comment by tseaver** [comment]
@busunkim96 literally **all** of my commits to Google's repositories over the last seven years have been signed with the `tseaver@palladion.com` email address.  The CLAbot seems flaky / fragile for commits which are merges of through-the-web suggestions.

**Pull request merged**
